packages/js/src/settings/store/pages.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable camelcase, complexity */
+import { createEntityAdapter, createSelector, createSlice } from "@reduxjs/toolkit";
+import apiFetch from "@wordpress/api-fetch";
+import { buildQueryString } from "@wordpress/url";
+import { map, trim } from "lodash";
+import { ASYNC_ACTION_NAMES, ASYNC_ACTION_STATUS } from "../constants";
+
+const pagesAdapter = createEntityAdapter();
+
+export const FETCH_PAGES_ACTION_NAME = "fetchPages";
+// Global abort controller for this reducer to abort requests made by multiple selects.
+let abortController;
+
+/**
+ * @param {Object} queryData The query data.
+ * @returns {Object} Success or error action object.
+ */
+export function* fetchPages( queryData ) {
+	yield{ type: `${ FETCH_PAGES_ACTION_NAME }/${ ASYNC_ACTION_NAMES.request }` };
+	try {
+		// Trigger the fetch pages control flow.
+		const pages = yield{
+			type: FETCH_PAGES_ACTION_NAME,
+			payload: { ...queryData },
+		};
+		return { type: `${ FETCH_PAGES_ACTION_NAME }/${ ASYNC_ACTION_NAMES.success }`, payload: pages };
+	} catch ( error ) {
+		return { type: `${ FETCH_PAGES_ACTION_NAME }/${ ASYNC_ACTION_NAMES.error }`, payload: error };
+	}
+}
+
+/**
+ * @returns {Object} The initial state.
+ */
+export const createInitialPagesState = () => pagesAdapter.getInitialState( {
+	status: ASYNC_ACTION_STATUS.idle,
+	error: "",
+} );
```

Comment:
I changed our setup a little in https://github.com/Yoast/wordpress-seo/pull/20359: https://github.com/Yoast/wordpress-seo/pull/20359/commits/ca30848c2e0d0db3e46b9978f75ffb192ba30591
See [Slack](https://yoast.slack.com/archives/C03RAL4QPAQ/p1685435892414319)
It might be nice to adjust to that here too 😇 
Plus having the slice `name` be a constant we use in the store index too.